neps/nep-0393.md

```diff
@@ -0,0 +1,542 @@
+
+This becomes especially important for proof-of-human stamps that can only be issued once per user. Few safeguards against misuse of SBT transfer (soul transfer or recovery) are contemplated:
+
+1. Users cannot recover an SBT by themselves. The issuer, a DAO or a smart contract (e.g. multisig) dedicated to manage the recovery should be assigned.
+2. Whenever a soul transfer is triggered then the SBT registry emits `SoulTransfer` and `Ban` events. The latter signals that the account can't host and receive any SBT in the future. It creates an inherit cost for such action: the account identity is burned.
+3. The recovery function is additional economical risk preventing account trading: user should always be able to recover his SBT, and move to another, not banned account.
+
+SBT recover MUST not trigger `SoulTransfer` nor `Ban`: malicious issuer could compromise the system by faking the token recovery and take over all other SBTs from a user. Only the owner of the account can make a Soul Transfer transaction and merge 2 accounts they owns.
+
+Soulbound tokens can have an _expire date_. This is useful for tokens which are related to real world certificates with expire time, or social mechanisms (e.g. community membership). Such tokens SHOULD have an option to be renewable. Examples include mandatory renewal with a frequency to check that the owner is still alive, or renew membership to a DAO that uses SBTs as membership gating.
+
+An issuer can provide an _sbt revocation_ in his contract (for example: when a related certificate or membership should be revoked, when an issuer finds out that there was an abuse or a scam, etc...). When doing so, the SBT registry MUST update token metadata and must emit the `Revoke` event. It's up to the registry to define revocation handling (either by burning a token, or changing expire date of it's metadata).
+
+A registry can emit a `Ban` event without doing soul transfer. Handling it depends on the registry governance or registry use cases. One example is to use social governance to identify fake accounts (like bots) - in that case the registry should allow to emit `Ban` and block a scam soul and block future transfers.
+NOTE: SBT smart contract (issuer) can have it's own list of blocked accounts or allowed only accounts.
+
+### Token Class
+
+SBT tokens can't be fractionized. Also, by definition there should be only one of a token per token class per user. Examples: user should not be able to receive few badges of the same class, or few proof of attendance to the same event.
+However we identify a need for having few token classes in a single contract:
+
+- badges: one contract with multiple badge class (community lead, OG...);
```

Comment:
I am not sure I understand it. On top of the class id there will be also badge id?

Reply:
I've updated the description to hopefully make it more clear.